feat: added and displayed pokemon overview

dapeventigan opened the pull request:
- added connector and vm
- added an overview page
- displayed pokemons using GridView
- added constants file for string, int, and custom values
- added a reusable capitalize first letter extention for lowercase strings
- removed homepage.dart

[Overview Page]
![image](https://user-images.githubusercontent.com/112401492/227420916-a39f2209-c409-468f-9f40-3fb63a7bb7e3.png)
[Flutter analyze and Dart formatter]
![image](https://user-images.githubusercontent.com/112401492/227421091-065e057c-3f2d-46e9-8158-bc5b24114efb.png)

dapeventigan committed:
feat: added and displayed pokemon overview

- added connector and vm
- added an overview page
- displayed pokemons using GridView
- added constants file for string, int, and custom values
- added a reusable capitalize first letter extention for lowercase strings

dapeventigan committed:
feat: added and displayed pokemon overview

- fixed merge conflicts
- removed home_page.dart

dapeventigan committed:
feat:added and displayed pokemon overview

- converted pokemon image url using string manipulation

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,28 @@
+  }) : super(key: key);
+
+  final List<Pokemon> pokemons;
+  @override
```
kindly add new line above this 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/widget/pokemon_overview_page_gridview.dart:
```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+
+class PokemonOverviewGridview extends StatelessWidget {
+  const PokemonOverviewGridview({
```
no need for this class, directly implement this inside your pokemon_overview_page 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/widget/pokemon_overview_page_gridview.dart:
```diff
@@ -0,0 +1,43 @@
+      itemCount: pokemons.length,
+      itemBuilder: (context, index) {
+        final pokemon = pokemons[index];
+        final pokemonIndex = index + numberOne;
```
why do you need to add 1 here? 🤔

shawn-ffufm commented on pokedex_async_redux/lib/features/widget/pokemon_overview_page_gridview.dart:
```diff
@@ -0,0 +1,43 @@
+        final pokemon = pokemons[index];
+        final pokemonIndex = index + numberOne;
+        final String pokemonImageUrlPath =
+            pokemonImageUri.replaceAll(RegExp(r'\[index\]'), '$pokemonIndex') + pokemonImageExtention;
```
u don't need regex here, a simple string manipulation will do, instead u might use that for getting pokemon id. look at jayvee's pr for reference on she implemented this part 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/widget/pokemon_overview_page_gridview.dart:
```diff
@@ -0,0 +1,43 @@
+        final pokemonIndex = index + numberOne;
+        final String pokemonImageUrlPath =
+            pokemonImageUri.replaceAll(RegExp(r'\[index\]'), '$pokemonIndex') + pokemonImageExtention;
+        return Card(
```
move this in a new stateless widget, maybe you can name it as `PokemonCard` 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/constants.dart:
```diff
@@ -0,0 +1,3 @@
+export 'custom_constant.dart';
```
the thing with this wild card export is that, when you import this constants.dart file then all of its components will be taken , which seems not efficient especially if I only need is the string_constant 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/custom_constant.dart:
```diff
@@ -0,0 +1,12 @@
+import 'package:flutter/material.dart';
+
+const appDefaultColor = Color.fromARGB(255, 13, 39, 83);
```
u can create a new dart file here then name it as `colors.dart` 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/custom_constant.dart:
```diff
@@ -0,0 +1,12 @@
+import 'package:flutter/material.dart';
+
+const appDefaultColor = Color.fromARGB(255, 13, 39, 83);
+const cardOverviewPadding = EdgeInsets.all(15);
```
seems this is not needed 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/custom_constant.dart:
```diff
@@ -0,0 +1,12 @@
+
+const appDefaultColor = Color.fromARGB(255, 13, 39, 83);
+const cardOverviewPadding = EdgeInsets.all(15);
+const crossAxisAlignmentCenter = CrossAxisAlignment.center;
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/utilities/custom_constant.dart:
```diff
@@ -0,0 +1,12 @@
+const appDefaultColor = Color.fromARGB(255, 13, 39, 83);
+const cardOverviewPadding = EdgeInsets.all(15);
+const crossAxisAlignmentCenter = CrossAxisAlignment.center;
+const pokemonNameFont = TextStyle(fontSize: 20);
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/utilities/custom_constant.dart:
```diff
@@ -0,0 +1,12 @@
+const pokemonNameFont = TextStyle(fontSize: 20);
+
+extension StringExtension on String {
+  String capitalize() {
```
see conversation [here](https://ffufmanilabootcamp.slack.com/archives/C027M886THS/p1679636361031749?thread_ts=1679635559.296549&cid=C027M886THS) about this 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/int_constant.dart:
```diff
@@ -0,0 +1,2 @@
+const numberOne = 1;
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/utilities/int_constant.dart:
```diff
@@ -0,0 +1,2 @@
+const numberOne = 1;
+const numberTwo = 2;
```
no need for this constant, just immediately use it based on where it's used 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utilities/string_constant.dart:
```diff
@@ -1,4 +1,9 @@
+import 'package:flutter/material.dart';
+
+const appBarTitle = Text("Pokedex");
```
declare the String value as a constant, not the widget 😅

shawn-ffufm commented on pokedex_async_redux/lib/utilities/string_constant.dart:
```diff
@@ -1,4 +1,9 @@
 const pokemonActionLimit = "151";
 const homePageTitle = "Home Page";
+const pokemonImageUri = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/home/[index]";
+const pokemonImageExtention = ".png";
```
u don't needed this based on what I said [here](https://ffufmanilabootcamp.slack.com/archives/C027M886THS/p1679636854575929?thread_ts=1679636686.472879&cid=C027M886THS) 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

dapeventigan committed:
feat: feat:added and displayed pokemon overview

- added dartX in pubspec and used it for capitalizing the first letter of pokemon name
- moved gridview back to overview page and created a widget for card instead
- fixed string manipulation for pokemon image url
- fixed constants
